packages/makeswift/src/integrate-next-app.ts

```diff
@@ -59,20 +59,89 @@ function addMakeswiftPages({ dir }: { dir: string }): void {
   const pagesFolder = getPagesFolder({ dir })
   const extension = getExtension({ dir })
 
+  function generateCatchAllRoute(extension: ReturnType<typeof getExtension>) {
+    switch (extension) {
+      case 'js':
+        return `import { Makeswift, Page as MakeswiftPage } from '@makeswift/runtime/next'
+
+export async function getStaticPaths() {
+  const makeswift = new Makeswift(process.env.MAKESWIFT_SITE_API_KEY)
+  const pages = await makeswift.getPages()
+
+  return {
+    paths: pages.map((page) => ({
+      params: {
+        path: page.path.split('/').filter((segment) => segment !== ''),
+      },
+    })),
+    fallback: 'blocking',
+  }
+}
+
+export async function getStaticProps(ctx) {
+  const makeswift = new Makeswift(process.env.MAKESWIFT_SITE_API_KEY)
+  const path = '/' + (ctx.params?.path ?? []).join('/')
+  const snapshot = await makeswift.getPageSnapshot(path, {
+    preview: ctx.preview,
+  })
+
+  if (snapshot == null) return { notFound: true }
+
+  return { props: { snapshot } }
+}
+
+export default function Page({ snapshot }) {
+  return <MakeswiftPage snapshot={snapshot} />
+}`
+      case 'ts':
```

Comment:
The extension here _has_ to be `tsx` otherwise there will be TypeScript errors with the JSX used in the file.

Reply:
Sorry, I think what I did is confusing here — this should be called `language`, not `extension`.

It is `${language}x` when I generate the file; I[ had caught that](https://github.com/makeswift/makeswift/pull/187/files#diff-038313253614925d50e16bd9d8bf678fbf73bd1e30bae3b4756ed381a12252e5R144). I think it's better if we operate on "language" than extension though, because some files will be `.jsx` and others `.js`, but for each file there's only 2 options — the JS or the TS variety. Whether it has React JSX is another dimension.

Reply:
Ah, got it. Yeah, we should probably go with the grain here and do what `create-next-app` does and have a boolean for whether we're using TypeScript or not.

Reply:
sounds good